changes/2020-06-16_required-examples/change.md

```diff
@@ -0,0 +1,98 @@
+
+This serves as a reference for all implementations that this change affects.
+
+| Language   | Repository                                                                            |
+| ---------- | ------------------------------------------------------------------------------------- |
+| Python     | [aws-encryption-sdk-c](https://github.com/aws/aws-encryption-sdk-python)              |
+| Java       | [aws-encryption-sdk-javascript](https://github.com/aws/aws-encryption-sdk-java)       |
+| C          | [aws-encryption-sdk-c](https://github.com/aws/aws-encryption-sdk-c)                   |
+| Javascript | [aws-encryption-sdk-javascript](https://github.com/aws/aws-encryption-sdk-javascript) |
+
+## Definitions
+
+### Conventions used in this document
+
+The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in
+[RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Summary
+
+This document defines the examples that each implementation MUST include.
+
+## Out of Scope
+
+The specific implementations details for each example
+are out of scope for this document.
+The template document for each example contains those details.
+
+## Motivation
+
+The two primary motivations for requiring examples are
+to provide guidance to AWS Encryption SDK customers
+and to force AWS Encryption SDK developers
+to use the APIs that they design and build
+to verify that they provide the intended user experience.
+
+## Drawbacks
+
+Requiring each implementation to include many examples
+increases the level of effort to create each implementation.
+We believe that the value to customers of
+improving the usability of APIs
+and providing guidance on how to use them
+justifies this cost.
+
+## Security Implications
+
+We MUST assume that any example will be used in production.
+Because of this, every example MUST only contain logic
+that is reasonable for production use
+unless explicitly identified otherwise.
+
+## Operational Implications
+
+Each implementation MUST include all examples
+in its continuous integration testing.
+
+## Guide-level Explanation
+
+Every implementation MUST include an example
+for each required use-case.
+These examples MUST be tested as part of
+the implementation's continuous integration testing.
+
+## Reference-level Explanation
+
+- Every example MUST be tested
+  as part of the implementation's continuous integration testing.
+- Every example MUST be independent and complete.
+- Every example SHOULD follow a consistent layout and framing.
+
+  - ex: Consistent function name and input parameters across examples.
+
+- Every example MUST only contain logic that is reasonable for production use.
+- If an example MUST contain logic that is not reasonable for production use,
+  it MUST include clear comments identifying that logic as such
+  and instructing the reader what they SHOULD do instead.
```

Comment:
Should the comments also identify the reason that the logic is not reasonable for production use?